app/post_persistent_notification.go

```diff
@@ -0,0 +1,380 @@
+		return nil
+	}
+
+	if posts, _, err := a.Srv().Store().PostPersistentNotification().Get(model.GetPersistentNotificationsPostsParams{PostID: post.Id}); err != nil {
+		return model.NewAppError("DeletePersistentNotificationsPost", "app.post_priority.delete_persistent_notification_post.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+	} else if len(posts) == 0 {
+		// Either the notification post already deleted or was never a notification post
+		return nil
+	}
+
+	if !*a.Config().ServiceSettings.AllowPersistentNotificationsForGuests {
+		user, nErr := a.Srv().Store().User().Get(context.Background(), loggedInUserID)
+		if nErr != nil {
+			var nfErr *store.ErrNotFound
+			switch {
+			case errors.As(nErr, &nfErr):
+				return model.NewAppError("DeletePersistentNotificationsPost", MissingAccountError, nil, "", http.StatusNotFound).Wrap(nErr)
+			default:
+				return model.NewAppError("DeletePersistentNotificationsPost", "app.user.get.app_error", nil, "", http.StatusInternalServerError).Wrap(nErr)
+			}
+		}
+		if user.IsGuest() {
+			c.Logger().Debug("DeletePersistentNotificationsPost: Persistent Notification feature is not enabled for guests.")
+			return nil
+		}
+	}
+
+	// if checkMentionedUser is "false" that would mean user is already authorized to delete the persistent-notification
+	// and the mention-validation can be skipped, so isUserMentioned will be "true".
+	isUserMentioned := !checkMentionedUser
+
+	// post owner is not allowed to stop the persistent notifications via ack, reply or reaction.
+	if checkMentionedUser && loggedInUserID != post.UserId {
+		if err := a.forEachPersistentNotificationPost([]*model.Post{post}, func(_ *model.Post, _ *model.Channel, _ *model.Team, mentions *ExplicitMentions, _ model.UserMap, _ map[string]map[string]model.StringMap) error {
+			if mentions.isUserMentioned(loggedInUserID) {
+				isUserMentioned = true
+			}
+			return nil
+		}); err != nil {
+			return model.NewAppError("DeletePersistentNotificationsPost", "app.post_priority.delete_persistent_notification_post.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+	}
+
+	if isUserMentioned {
+		if err := a.Srv().Store().PostPersistentNotification().Delete([]string{post.Id}); err != nil {
+			return model.NewAppError("DeletePersistentNotificationsPost", "app.post_priority.delete_persistent_notification_post.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+	}
+
+	return nil
+}
+
+func (a *App) SendPersistentNotifications() error {
+	notificationInterval := time.Duration(*a.Config().ServiceSettings.PersistentNotificationIntervalMinutes) * time.Minute
+	notificationMaxCount := int16(*a.Config().ServiceSettings.PersistentNotificationMaxCount)
+
+	// fetch posts for which first notificationInterval duration has passed
+	maxCreateAt := time.Now().Add(-notificationInterval).UnixMilli()
+
+	pagination := model.CursorPagination{
```

Comment:
I don't see any place where `CursorPagination.Direction` is used with other value different than `down`. Does it make sense to include it in the struct at all? We are adding complexity without getting any benefit (we can change it later, but I don't see any reason to get the notifications in a different order).

Reply:
I've tried to make it a generic model, so that it can be used/extended later on without thinking of the direction logic.